core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -228,6 +235,25 @@ public Builder httpExecutor(HttpExecutor val) {
             this.httpExecutor = val;
             return this;
         }
+        public Map<String, String> buildBaseHeaders() {
+            if (Boolean.TRUE.equals(preemptiveBasicAuth)) {
+                Credentials creds = credentials;
+                if (creds == null) {
+                    throw new IllegalArgumentException("Must not enable preemptiveBasicAuth when there are no credentials, in feed for "+baseUri);
```

Comment:
We do this a lot in brooklyn and I'm not sure why.  Given that we're using a typed language why not create a MisconfiguredPreemptiveAuthException?  We can check the type of the exception thrown in tests and a tweak to the error text doesn't break the tests.